gradoop-flink/src/main/java/org/gradoop/flink/model/api/epgm/LogicalGraph.java

```diff
@@ -745,4 +748,45 @@ public GraphCollection splitBy(String propertyKey) {
   public void writeTo(DataSink dataSink) throws IOException {
     dataSink.write(this);
   }
+
+  /**
+   * Prints the GDL formatted graph to the standard output.
+   *
+   * @throws Exception
+   */
+  public void print() throws Exception {
+    List<Vertex> vertices = getVertices().collect();
+    List<Edge> edges = getEdges().collect();
+    GraphHead graphHead = getGraphHead().collect().get(0);
```

Comment:
Using `.collect()` will call `ExecutionEnvironment.execute()`. Your code will run the flink pipeline 3 times.
Please use the `LocalCollectionOutputFormat` instead.

Reply:
@p-f is absolute right. Each `collect( )` will be a separate flink job. Maybe you can introduce a `GDLElement` where you store all necessary information and `collect( )` once to print. 

```
DataSet<GDLElement> vertexToPrint = graph.getVertices().map(new ToGDLElement);
DataSet<GDLElement> edgeToPrint = graph.getEdges().map(new ToGDLElement);
DataSet<GDLElement> headToPrint = graph.getGraphHead().map(new ToGDLElement);

information = VertexToPrint.union(edgeToPrint).union(...);

List<GDLElement> stuffToPrint = information.collect( );
```
You should store the entity type of each element within `GDLElement`, with that you can distinguish between each element and print your GDL string.